libs/sdk-bindings/src/breez_sdk.udl

```diff
@@ -133,7 +133,7 @@ dictionary RouteHint {
 };
 
 dictionary LNInvoice {
-    string bolt11;
+    string raw_invoice;
```

Comment:
Why is this field changed? do we use it for bolt12 also? I couldn't find any reference for such usage in the code.

Reply:
We should consider whether the `LNInvoice` wrapper should be able to support both Bolt11 and Bolt12 fields at once, given the overlap between the two. Currently CLN makes the invoice field naming distinct between the two types in most methods (e.g. createinvoice, listinvoices etc.) with the exception of outward methods like sendpay and pay. 

The initial reasoning was that, in terms of UX, I believe that the final user is probably not really interested in the type of invoice they are paying (whether bolt11 or 12), so perhaps unifying the optional `bolt11` and `bolt12` parameters into a single field `raw_invoice` could make sense and reduce the need to constantly check which one it is. Additionally, the client would know the type a priori to the fetchinvoice call simply by the scanned QR (offer or bolt11 invoice), therefore there is most probably a way to make this work with a single field.

On the other hand, I can also see instances where the developers would like to know which type of invoice it is at any given time. That is of course more flexible on the developer's side as they are now able to handle the two distinct cases better (such as adding more fields depending on which kind it is [e.g. quantity]) and throw an error, for example, if neither is specified, without having to refer to the original offer to discern so. 

I'll probably roll back this change and specify the two fields as optional, just wanted to let you in on what the thought process was behind that decision

Reply:
Thanks for the details on what you had in mind @hydra-yse 
For me it is even more simple than that. As long as the code doesn't user LNINvoice for bolt12 (and we don't even expose this parsing to the user) then there is no reason to unified.
If/When we implement  and expose such parsing we will have more information to take the right decision.

Reply:
Agree with Roei. For parsing we can differentiate between bolt11 and bolt12 invoices. For send_payment we can call the `bolt11` argument `payment_request`. The developer would call `send_payment` with either the bolt11 fields from LNInvoice or the `bolt12_offer` field from LNOffer.